examples/list-binding-pattern/list_binding_pattern.bal

```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
+
+public function main() {
+
+    // This will create 3 variables named `id`, `firstname`, and `lastname` with the following types.
+    [int, [string, string]] [id, [firstname, lastname]] = getDetails();
```

Comment:
Shall we add one usage of `wildcard binding pattern` inside list BP

Reply:
Since we have a separate BBE for `wildcard binding pattern`, won't it complicate things here?

Reply:
`wildcard binding pattern` inside `list binding pattern` is useful. IMO it's good to mention it here since we don't have such an example in `wildcard binding pattern` BBE as well.

Reply:
Done.